src/main/java/mate/academy/dao/impl/BookDaoImpl.java

```diff
@@ -0,0 +1,114 @@
+package mate.academy.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.BookDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String insertRequest = "INSERT INTO books (title,price)"
+                + " VALUES (?,?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createBookStatement =
+                        connection.prepareStatement(insertRequest,
+                                Statement.RETURN_GENERATED_KEYS)) {
+            createBookStatement.setString(1, book.getTitle());
+            createBookStatement.setBigDecimal(2, book.getPrice());
+            createBookStatement.executeUpdate();
+            ResultSet generatedKeys = createBookStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Can`t create insert format to DB", e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String findByIdRequest = "SELECT * FROM books WHERE id = ?";
```

Comment:
Your find/findAll/update operations shouldn't work with deleted records in DB
Please expand conditions in queries to avoid getting/updating records that were soft-deleted